langchain/src/vectorstores/azuresearch.ts

```diff
@@ -0,0 +1,541 @@
+      return this.hybridSearchVectorWithScore(
+        query,
+        await this.embeddings.embedQuery(query),
+        k,
+        filter
+      );
+    } else if (searchType === "semantic_hybrid") {
+      return this.semanticHybridSearchVectorWithScore(
+        query,
+        await this.embeddings.embedQuery(query),
+        k,
+        filter
+      );
+    }
+
+    throw new Error(`Unrecognized search type '${searchType}'`);
+  }
+
+  /**
+   * Perform a hybrid search using text search.
+   *
+   * @param query
+   * @param queryVectors
+   * @param k
+   * @param filter
+   */
+  async hybridSearchVectorWithScore(
+    query: string,
+    queryVectors: number[],
+    k: number,
+    filter?: string
+  ): Promise<[Document, number][]> {
+    const { results } = await this.client.search(query, {
+      vectors: [{
+        value: queryVectors,
+        fields: [DEFAULT_FIELD_CONTENT_VECTOR],
+        kNearestNeighborsCount: k,
+      }],
+      filter,
+      top: k,
+    });
+
+    const docsWithScore: [Document, number][] = [];
+
+    for await (const item of results) {
+      const document = new Document({
+        pageContent: item.document.content,
+        metadata: item.document.metadata,
+      });
+
+      docsWithScore.push([document, item.score]);
+    }
+
+    return docsWithScore;
+  }
+
+  /**
+   * Perform a hybrid search using Semantic configuration.
+   *
+   * @param query
+   * @param queryVectors
+   * @param k
+   * @param filter
+   */
+  async semanticHybridSearchVectorWithScore(
+    query: string,
+    queryVectors: number[],
+    k: number,
+    filter?: string
+  ): Promise<[Document, number][]> {
+    const { results } = await this.client.search(query, {
+      vectors: [{
+        value: queryVectors,
+        fields: [DEFAULT_FIELD_CONTENT_VECTOR],
+        kNearestNeighborsCount: k,
+      }],
+      filter,
+      top: k,
+      queryType: "semantic",
+      queryLanguage: this.params.search.language ?? "en-us",
+      semanticConfiguration: this.params.search.semantic ?? "default",
+      captions: "extractive",
+      answers: "extractive",
+    });
+
+    const docsWithScore: [Document, number][] = [];
+
+    for await (const item of results) {
+      const document = new Document({
+        pageContent: item.document.content,
+        metadata: item.document.metadata,
+      });
+
+      docsWithScore.push([document, item.rerankerScore ?? item.score]);
+    }
+
+    return docsWithScore;
+  }
+
+  /**
+   * Execute a vector similarity search.
+   *
+   * @param query
+   * @param k
+   * @param filter
+   */
+  async similaritySearchVectorWithScore(
+    query: number[],
+    k: number,
+    filter?: string
+  ): Promise<[Document, number][]> {
+    const { results } = await this.client.search("", {
+      vectors: [{
+        value: query,
+        fields: [DEFAULT_FIELD_CONTENT_VECTOR],
+        kNearestNeighborsCount: k,
+      }],
+      filter,
```

Comment:
like the other search methods, needs to have a `top: k` field passed in

Reply:
When search using vector similarity the parameter to limit the results is `kNearestNeighborsCount`.